modules/effects/spec/effects_error_handler.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { ErrorHandler, Type } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import { Action, Store } from '@ngrx/store';
+import { Observable, of } from 'rxjs';
+import {
+  createEffect,
+  EFFECTS_ERROR_HANDLER,
+  EffectsModule,
+  EffectSources,
+} from '..';
+
+describe('NgRx Effects Error Handler spec', () => {
+  let subscriptionCount: number;
+  let globalErrorHandler: jasmine.Spy;
+  let dispatch: jasmine.Spy;
+  let storeNext: jasmine.Spy;
+
+  function makeEffectTestBed(...providers: any[]) {
+    subscriptionCount = 0;
+
+    TestBed.configureTestingModule({
+      imports: [EffectsModule.forRoot([ErrorEffect])],
+      providers: [
+        {
+          provide: Store,
+          useValue: {
+            dispatch: jasmine.createSpy('dispatch'),
+            next: jasmine.createSpy('storeNext'),
+          },
+        },
+        {
+          provide: ErrorHandler,
+          useValue: {
+            handleError: jasmine.createSpy('globalErrorHandler'),
+          },
+        },
+        ...providers,
+      ],
+    });
+
+    TestBed.get(EffectSources);
+    globalErrorHandler = TestBed.get(ErrorHandler).handleError;
+    const store = TestBed.get(Store);
+    dispatch = store.dispatch;
+    storeNext = store.next;
+  }
+
+  it('should retry and notify error handler when effect error handler is not provided', () => {
+    makeEffectTestBed();
+
+    expect(subscriptionCount).toBe(2);
+    expect(globalErrorHandler).toHaveBeenCalledWith(new Error('effectError'));
+  });
+
+  it('should use custom error behavior when EFFECTS_ERROR_HANDLER is provided', () => {
+    const effectsErrorHandlerSpy = jasmine.createSpy();
+    effectsErrorHandlerSpy.and.callFake((_$, errorHandler: ErrorHandler) => {
+      errorHandler.handleError(new Error('inside custom handler'));
+
+      return of({ type: 'custom action' });
+    });
+
+    makeEffectTestBed({
+      provide: EFFECTS_ERROR_HANDLER,
+      useValue: effectsErrorHandlerSpy,
+    });
+
+    expect(effectsErrorHandlerSpy).toHaveBeenCalledWith(
+      jasmine.any(Observable),
+      TestBed.get(ErrorHandler)
+    );
+    expect(globalErrorHandler).toHaveBeenCalledWith(
+      new Error('inside custom handler')
+    );
+    expect(subscriptionCount).toBe(0);
+    expect(storeNext).toHaveBeenCalledWith({ type: 'custom action' });
```

Comment:
I find this assertion to be irrelevant for what you are putting under test

Reply:
Should I break it into another `it()` block? I think it is important to test that the value returned from the custom error handler is what ends up being passed to the store as an action stream.

I've updated the error handler substitution to make this clearer